float32 width
float32 max_width
bool is_grasped
uint16 temperature
uint64 duration_ms